### PR TITLE
Refactor and update input configuration.

### PR DESCRIPTION
**Pull request by josiah-tesfu**
Updated the input configuration. Configuration is initially set through the editor. Once the node is deployed, configuration can be set dynamically from input. If any parameters from the input message are empty, the previously set configurations are used.

Refactored the code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,34 +22,35 @@ Assigned ports for RTD measurements:
...
 
 ## Inputs
 
-- Any message can be used to trigger this node.
+- `payload` (_number_)<br>
+  The input selection -- the channel or channel diff depending on the read type.
+- `read` (_string_)<br>
+  'voltage', 'differential', or 'rtd'.
+- `dataRate` (_string_)<br>
+- `adc` (_string_)<br>
```
Comment (farzadpanahi) [inline README.md]: ## Inputs
- `payload` (_number_)
  The input selection -- the channel number (1 to 8) or differential channels number (1 to 4) depending on the read type. Ignored for read type `rtd`.
- `readType` (_string_)
  `single`, `diff`, or `rtd`. Default is `single`.
- `dataRate` (_number_)
  ADC data rate. 
  Valid values for ADC 1: <values>. Default is <default>.
  Valid values for ADC 2: <values>. Default is <default>.
- `adc` (_number_)
  ADC number to use for reads. Valid values are 1 and 2. Forced to 2 for read type `rtd`. Default is 1.

File: README.md [hunk 2]
```diff
@@ -22,34 +22,35 @@ Assigned ports for RTD measurements:
...
+    Voltage reading of the specified differential.
 - When configured to read RTD:
-  - `payload` (*number*)<br>
-  RTD temperature reading.
-
-
-**NOTE:** Currently, EdgePi nodes are only available on x86 systems.
-
+  - `payload` (_number_)<br>
+    RTD temperature reading.
```
Comment (farzadpanahi) [inline README.md]: ## Outputs

- Read type `single`:
  - `payload` (_number_)<br>
    Voltage reading of the specified single-ended analog input channel.
- Read type `diff`:
  - `payload` (_number_)<br>
    Differential voltage reading of the specified differential channels. 
- Read type `rtd`:
  - `payload` (_number_)<br>
    RTD temperature reading in Celsius.

Comment (farzadpanahi) [inline README.md]: ## Outputs

- Read type `single`:
  - `payload` (_number_)
    Voltage reading of the specified single-ended analog input channel.
- Read type `diff`:
  - `payload` (_number_)
    Differential voltage reading of the specified differential channels. 
- Read type `rtd`:
  - `payload` (_number_)
    RTD temperature reading in Celsius.

File: edgepi-adc.html [hunk 3]
```diff
@@ -1,251 +1,279 @@
...
-      const dataRateField = document.querySelector(".form-row.data-rate")
+      const read = document.getElementById("node-input-read");
+      const adcNum = document.getElementById("node-input-adcNum");
+      const adc1 = document.getElementById("adc-option-adc1");
+      const dataRateField = document.querySelector(".form-row.data-rate");
       const adc1DataRate = document.getElementById("node-input-adc1DataRate");
       const adc2DataRate = document.getElementById("node-input-adc2DataRate");
-			const channelField = document.querySelector(".form-row.channel");
+      const channelField = document.querySelector(".form-row.channel");
       const diffField = document.querySelector(".form-row.diff");
```
Comment (farzadpanahi) [inline edgepi-adc.html:35]: make sure these fields match what we have in the readme.

**Review by farzadpanahi — CHANGES_REQUESTED** [review]
let's fix the inputs and ouputs and make sure they work as expected.

File: README.md (hunk 2 above)
Comment (jimmy121192) [reply README.md]: Updated

Comment (jimmy121192) [reply README.md]: Updated

File: edgepi-adc.html (hunk 3 above)
Comment (jimmy121192) [reply edgepi-adc.html:35]: Updated

File: README.md (hunk 1 above)
Comment (jimmy121192) [reply README.md]: Updated

**Comment by jimmy121192** [comment]
Updated content, and tested inputs and outputs.
Screen Shot 2024-01-11 at 12 55 00 PM

File: README.md [hunk 4]
```diff
@@ -36,12 +36,16 @@ Assigned ports for RTD measurements:
...
-- `dataRate` (_string_)<br>
-- `adc` (_string_)<br>
+- **payload** (_number_)<br>
+  The input selection -- the channel number (1 to 8) or differential channels number (1 to 4) depending on the read type. Ignored for read type `rtd`.
+- **readType** (_string_)<br>
+  `single`, `diff`, or `rtd`. Default is `single`.
+- **dataRate** (_number_)<br>
+  ADC data rate.
+  Valid values for ADC 1 . Default is `38400`.
+  Valid values for ADC 2 . Default is `800`.
```
Comment (farzadpanahi) [inline README.md]: list the valid values!

File: edgepi-adc.js [hunk 5]
```diff
@@ -5,20 +5,19 @@ module.exports = async function (RED) {
...
-            adcNum = msg.adc ?? adcNum;
+            read = msg.readType || read; //prioritize the inputs in the msg
+            adcNum = msg.adc || adcNum;
             inputSelection = msg.payload || inputSelection;
-            dataRate = msg.dataRate ?? dataRate;
-
+            dataRate = msg.dataRate || dataRate;
             if (read !== "rtd") {
               await adc.setRtd(false, adcNum - 1);
             }
```
Comment (farzadpanahi) [inline edgepi-adc.js]: I gave it another thought. I think we should force adc 2 when rtd is enabled. so change to:
```
await adc.setRtd(false, 1);
```

**Review by farzadpanahi — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -22,34 +22,39 @@ Assigned ports for RTD measurements:
...
+- **adc** (_number_)<br>
+  ADC number to use for reads. Valid values are **1** and **2**. Default is **1**.
```
Comment (farzadpanahi) [inline README.md]: add back: "Forced to 2 when RTD `readType` is `rtd`"

**Review by farzadpanahi — CHANGES_REQUESTED** [review]

File: edgepi-adc.js (hunk 5 above)
Comment (jimmy121192) [reply edgepi-adc.js]: updated

File: README.md (hunk 6 above)
Comment (jimmy121192) [reply README.md]: added

File: README.md (hunk 4 above)
Comment (jimmy121192) [reply README.md]: done

**Review by farzadpanahi — APPROVED** [review]

**Pull request merged**